src/components/canvas-tools/text-tool.tsx

```diff
@@ -40,10 +40,20 @@ export default class TextToolComponent extends BaseComponent<IProps, IState> {
     const { readOnly, model } = this.props;
     const { content } = model;
     const { ui } = this.stores;
-    const op = change.operations.get(0);
-    if (op.type === "set_selection") {
-      ui.setSelectedTile(model);
+
+    // determine last focus state from list of operations
+    let isFocused: boolean | undefined;
+    change.operations.forEach(op => {
+      if (op && op.type === "set_selection") {
+        isFocused = op.selection.get("isFocused");
```

Comment:
Nice! This allows de-selecting text boxes, though graph tiles still can't be deselected by clicking on nothing

Reply:
Since JSXGraph has no notion of focus for the board (as far as I know) we can't use JSXGraph notifications for this the way we can with Slate. The solution for clicking outside deselecting geometry (and other similar) tiles is to have a click handler at the DocumentContentComponent, which I decided to leave as a separate (1-point) story.